src/main/java/com/rtb/projectmanagementtool/user/UserProfileServlet.java

```diff
@@ -109,7 +109,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
 
     // get parameters
-    TaskData task = (TaskData) request.getParameter("task")
+    TaskData task = (TaskData) request.getAttribute("task");
```

Comment:
this is null even though I thought I was passing it from user-private-comments.jsp. the task (which I printed out in the jsp) is not null, so this is just not being passed.

Reply:
```
<%request.setAttribute("task", task);%>
```
I'm not entirely sure if this is the cause, but this line in the jsp is not inside of the form. Might be why it's null. Instead of trying to parse TaskData from the "task" parameter, and since you've already sent in the taskID, you could use taskController to get the TaskData

Reply:
You don't have an input with name="task" in your form, so you wouldn't get an attribute with name "task" in your request.  Even if you add the taskData as an input in your form, HTTP requests will only send you string data, which means you'll need to convert to taskData somehow.

It also might not be a good idea to pass the whole task in the request if you don't need the whole task (eg. you probably don't need the task description).  See if you can instead just pass the taskID into the request.